src/zenml/integrations/gcp/secrets_manager/gcp_secrets_manager.py

```diff
@@ -105,14 +297,15 @@ def register_secret(self, secret: BaseSecretSchema) -> None:
         Raises:
             SecretExistsError: if the secret already exists
         """
+        self.validate_secret_name_or_namespace(secret.name)
         self._ensure_client_connected()
 
-        if secret.name in self.get_all_secret_keys():
+        if not self._list_secrets(secret.name):
```

Comment:
Shouldn't this be without the `not` in order to work?

Reply:
Curiously enough, I just ran the integration tests and they also caught this same issue, but they passed just fine the last time I ran them. Probably something that slipped during a merge. Thanks for noticing !